### PR TITLE
INT-3659 Add InputStream support to FileWritingMessageHandler

### PR DESCRIPTION
**Pull request by ghillert**
- Add tests
- Add documentation

https://jira.spring.io/browse/INT-3659

### CODE REVIEW

File: spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java [hunk 1]
```diff
@@ -355,19 +358,32 @@ private File retrieveOriginalFileFromHeader(Message<?> message) {
...
+			if (logger.isInfoEnabled()) {
+				logger.info(String.format("Failed to move file '%s'. Using copy and delete fallback.",
+						sourceFile.getAbsolutePath()));
+			}
+		}
+		final BufferedInputStream bis = new BufferedInputStream(new FileInputStream(sourceFile));
+		return handleInputstreamMessage(bis, sourceFile, tempFile, resultFile);
+	}
+
+	private File handleInputstreamMessage(final InputStream sourceFileInputstream, File originalFile, File tempFile, final File resultFile) throws IOException {
```
Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java]: Method name and first argument typo: `...InputStream...`

Comment (artembilan) [inline spring-integration-file/src/main/java/org/springframework/integration/file/FileWritingMessageHandler.java]: In addition bad code line length - 120 is preferable.

**Comment by artembilan** [comment]
OK. All concerns are minor: fixing during merge...

**Comment by artembilan** [comment]
Merged as https://github.com/spring-projects/spring-integration/commit/247232bdde24b81814a82100743f77d881aaf06b

**Comment by ghillert** [comment]
Thanks a million!

**Pull request closed without merging**